Overriding `quarkus.package.output-directory` can cause random native build failures

### Describe the bug

Overriding `quarkus.package.output-directory` can lead to random failures when building native image, for example:

```
Caused by: java.nio.file.NoSuchFileException: /Users/galder/1/sample/sample-it/target/runner-native/sample-it-1.0.0-SNAPSHOT-native-image-source-jar/sample-it-1.0.0-SNAPSHOT-runner.jar
    at sun.nio.fs.UnixException.translateToIOException (UnixException.java:92)
    at sun.nio.fs.UnixException.rethrowAsIOException (UnixException.java:106)
    at sun.nio.fs.UnixException.rethrowAsIOException (UnixException.java:111)
    at sun.nio.fs.UnixFileAttributeViews$Posix.readAttributes (UnixFileAttributeViews.java:257)
    at sun.nio.fs.UnixFileAttributeViews$Posix.readAttributes (UnixFileAttributeViews.java:168)
    at jdk.nio.zipfs.ZipFileSystem.getPosixAttributes (ZipFileSystem.java:1891)
    at jdk.nio.zipfs.ZipFileSystem.sync (ZipFileSystem.java:1778)
    at jdk.nio.zipfs.ZipFileSystem.lambda$close$9 (ZipFileSystem.java:484)
    at java.security.AccessController.doPrivileged (AccessController.java:571)
    at jdk.nio.zipfs.ZipFileSystem.close (ZipFileSystem.java:483)
    at io.quarkus.deployment.pkg.steps.JarResultBuildStep.buildNativeImageThinJar (JarResultBuildStep.java:1025)
    at io.quarkus.deployment.pkg.steps.JarResultBuildStep.buildNativeImageJar (JarResultBuildStep.java:983)
```

I've added some extra logging (I'll send a separate PR for the logging) that shows that when `quarkus.package.output-directory` is overriden, `JarResultBuildStep#buildNativeImageJar` and `JarResultBuildStep#buildRunnerJar` get in a race to empty overlapping folders:

```
[DEBUG] [io.quarkus.builder] (Thread[#70,build-32,5,build group]) Starting step "io.quarkus.deployment.pkg.steps.JarResultBuildStep#buildRunnerJar"
[DEBUG] [io.quarkus.builder] (Thread[#41,build-3,5,build group]) Starting step "io.quarkus.deployment.pkg.steps.JarResultBuildStep#buildNativeImageJar"

[DEBUG] [io.quarkus.bootstrap.util.IoUtils] (Thread[#41,build-3,5,build group]) Create or empty directory /Users/galder/1/sample/sample-it/target/runner-native/sample-it-1.0.0-SNAPSHOT-native-image-source-jar
[DEBUG] [io.quarkus.bootstrap.util.IoUtils] (Thread[#70,build-32,5,build group]) Create or empty directory /Users/galder/1/sample/sample-it/target/runner-native
```

The 2 threads are competing to empty the same folders (e.g. anything under `sample-it-1.0.0-SNAPSHOT-native-image-source-jar` is overlapping) and sometimes they get on each others way and results in a build failure and half emptied folder.

When `quarkus.package.output-directory` is not overriden, we can see that the folders are not overlapping:

```
[DEBUG] [io.quarkus.builder] (Thread[#42,build-10,5,build group]) Starting step "io.quarkus.deployment.pkg.steps.JarResultBuildStep#buildRunnerJar"
[DEBUG] [io.quarkus.builder] (Thread[#39,build-7,5,build group]) Starting step "io.quarkus.deployment.pkg.steps.JarResultBuildStep#buildNativeImageJar"
[DEBUG] [io.quarkus.bootstrap.util.IoUtils] (Thread[#39,build-7,5,build group]) Create or empty directory /Users/galder/1/mendrugo/getting-started-reactive/quickstart/target/getting-started-reactive-1.0.0-SNAPSHOT-native-image-source-jar
...
[DEBUG] [io.quarkus.bootstrap.util.IoUtils] (Thread[#42,build-10,5,build group]) Create or empty directory /Users/galder/1/mendrugo/getting-started-reactive/quickstart/target/quarkus-app
```

I'm unsure what's the fix but the sharing of the folder is the result of this in `JarResultBuildStep.outputTarget`:

```
        Path path = packageConfig.outputDirectory.map(s -> bst.getOutputDirectory().resolve(s))
                .orElseGet(bst::getOutputDirectory);
```

--- a/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/AppCDSBuildStep.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/AppCDSBuildStep.java
@@ -49,7 +49,7 @@ public class AppCDSBuildStep {
         producer.produce(new AppCDSRequestedBuildItem(outputTarget.getOutputDirectory().resolve("appcds")));
     }
 
-    @BuildStep
+    @BuildStep(onlyIfNot = NativeOrNativeSourcesBuild.class)
     public void build(Optional<AppCDSRequestedBuildItem> appCDsRequested,
             JarBuildItem jarResult, OutputTargetBuildItem outputTarget, PackageConfig packageConfig,
             CompiledJavaVersionBuildItem compiledJavaVersion,

